### ISSUE TITLE
Able to create two accounts with the same username

### ISSUE THREAD

**Issue by jpgklassen** [clipperz/password-manager#116]
I'm able to create two Clipperz accounts with the same username and different passwords. Let's call them Account A and Account B. If I change the password of Account B to match the password of Account A, I am told that the password change is successful. Then, when I sign in with the new password, I get logged into Account A.

This is a big security issue!!!!

**Comment by dschenck** [comment]
+1

**Comment by KubaO** [comment]
There's some fundamental misunderstanding here. **If anyone can guess the password, an account is owned, end of story.**

The correct interpretation of your "problem" is not that you get "logged into Account A", but that maybe you lost data from Account B (if that actually did happen vs. the accounts merging). But only you're supposed to know the password to *YOUR* accounts. If I can already guess the password to *YOUR* Account A, then the game is over, and it doesn't matter that there even is an Account B. Make sure nobody can guess it and that's all.

Other applications give some sort of false sense of security by allowing passwords to be duplicated between user accounts - when in fact **it should never ever not in the life of the entire Universe be the case that two users have the same password**. Why? Because such passwords *are by definition unsafe*!! If you can guess it in a person's lifetime, it's a lousy password. Use a better one. Full stop.

Let me rephrase: if someone else has *your password*, it doesn't matter that they also made it "their own". They have your password. They may not know your username *yet*. But the assumption is that it will happen. So please - don't ever use a password that you imagine anyone could guess, ever.

**Comment by KubaO** [comment]
I'd say this issue should be closed as is, or reformulated to indicate a problem with data loss if in fact Account B, after merging with Account A, loses its data and only Account A's data is retained.

**Comment by marcor** [comment]
> There's some fundamental misunderstanding here. **If anyone can guess the password, an account is owned, end of story.**

I think I agree with you KubaO. Still, the fundamental misunderstanding is comprehensible as long as both a username and a password are required. Unless there is another technical reason for this separation, the security risks you describe would be more transparent if the user was only required to enter a passphrase to login. That way, there would be no misunderstanding and the interface could never behave in unexpected and unintended ways (which is the problem here).

In other words: what is the purpose of a username, if not that of being unique? And if has not even that purpose, why not get rid of it entirely?

On the other hand, if a username has a purpose, and it is not meant to be unique: is something like the above happened, it would not be okay to merge the contents of the two accounts: on the contrary,  I'd like the system to lock down all access to account A, and prevent user B from changing the password to account B. In that scenario, user B would now know the password of user A (which is compromised), but they would not be able access the contents of account A, and there would not be any data loss. Is this feasible in Clipperz?

**Comment by gcsolaroli** [comment]
I'll try to provide the full context to share why Clipperz implementation is the way it is, and why we do currently not have any plan to change it:
- username and password is the most common access pattern to online services, and even if our implementation definitely doesn't look like the average one, we wanted to provide a familiar UX to our users;
- the choice to identify users with a value computed "salting" the username with the password prevents any possible identification of the account data knowing the username; if anyone want to get access to a specific account data, they will not be able to focus their attention to just a specific subset of data, and they will have to process all the data;
- it make trivially easy to create "ghost" accounts that could provide a plausible alternative in case someone is faced to the request to open up their Clipperz account using the wrench system (https://xkcd.com/538/).
